src/ImageSharp/PixelFormats/PixelOperations{TPixel}.cs

```diff
@@ -83,7 +83,7 @@ public partial class PixelOperations<TPixel>
         /// <param name="configuration">A <see cref="Configuration"/> to configure internal operations</param>
         /// <param name="sourcePixels">The <see cref="Span{T}"/> to the source colors.</param>
         /// <param name="destVectors">The <see cref="Span{T}"/> to the destination vectors.</param>
-        internal virtual void ToVector4(
+        public virtual void ToVector4(
```

Comment:
This method should not be virtual.

Reply:
Wait, this method is actually overridden by some pixel-format-specific implementations of `PixelOperations<TPixel>`, like for [`RgbaVector`](https://github.com/Sergio0694/ImageSharp/blob/e98c0ff46bc8fd21a91f74b61bc28484a6f2c0b3/src/ImageSharp/PixelFormats/PixelImplementations/RgbaVector.PixelOperations.cs#L37), [`Rgba32`](https://github.com/Sergio0694/ImageSharp/blob/e98c0ff46bc8fd21a91f74b61bc28484a6f2c0b3/src/ImageSharp/PixelFormats/PixelImplementations/Rgba32.PixelOperations.cs#L23) and others. It's also in the `GenerateRgba32CompatibleVector4ConversionMethods` part in the `_Common.ttinclide` file, [here](https://github.com/Sergio0694/ImageSharp/blob/e98c0ff46bc8fd21a91f74b61bc28484a6f2c0b3/src/ImageSharp/PixelFormats/PixelImplementations/Generated/_Common.ttinclude#L139).

I don't think we can remove that `virtual` right now, given the rest of the codebase as it is now. That is, unless you have some refactoring in mind to make that possible 🤔

Reply:
Yeah, we're gonna have to have a think about this!

Reply:
@Sergio0694 I did not have the chance to check the code, but if that method is overriden, it's a mistake. All scenarios shall be covered with a single virtual overload which is taking the most arguments, the shorter overloads shall be non-virtual wrappers only.

Otherwise everything else LGTM, if someone could confirm that T4 generation is  working.

Reply:
@antonfirsov So, I'm not sure how to proceed here. From what I see, even before my changes in this branch, that `virtual ToVector4` method was already being overridden in virtually all the pixel-format specific `PixelOperations<TPixel>` classes.
For instance, this happens both in the explicit [`RgbaVector`](https://github.com/Sergio0694/ImageSharp/blob/4427e2467a3f46e34b9059c99304266aea17786b/src/ImageSharp/PixelFormats/PixelImplementations/RgbaVector.PixelOperations.cs#L37) and [`Rgba32`](https://github.com/Sergio0694/ImageSharp/blob/4427e2467a3f46e34b9059c99304266aea17786b/src/ImageSharp/PixelFormats/PixelImplementations/Rgba32.PixelOperations.cs#L23) types, as well as in all the generated pixel-format classes (eg. here in [`Argb32.Generated`](https://github.com/Sergio0694/ImageSharp/blob/4427e2467a3f46e34b9059c99304266aea17786b/src/ImageSharp/PixelFormats/PixelImplementations/Generated/Argb32.PixelOperations.Generated.cs#L51)).

I'm not sure how to proceed here if you want to remove the `virtual` modifier: as far as I can see, right now each pixel-format is [returning an instance of its own `PixelOperations<TPixel>` private class](https://github.com/Sergio0694/ImageSharp/blob/4427e2467a3f46e34b9059c99304266aea17786b/src/ImageSharp/PixelFormats/PixelImplementations/Rgba32.cs#L258), exposed as `PixelOperations<TPixel>`, so that when the methods are called on that instance, the right overload for the specific pixel-format is called. And I mean, this makes sense, and it looks great. My question is: if you want to remove the `virtual` modifier from the default implementation of those methods in the base `PixelOperations<TPixel>` class, how can we then have each pixel-format call its own custom version of those APIs?

`PixelOperations<TPixel>` is not `abstract` so we can't even just set that method as `abstract` from there and have each pixel-format specific class implement it. Plus as far as I know there are some pixel-formats that do fallback on that base implementation, so this wouldn't work either.

I'm not sure how to proceed, I might very well be missing something here, I'm confused 🤔

Reply:
@Sergio0694 I think you were looking at the wrong method. The one highlighted by @antonfirsov  does not have any overloads. I've removed the `virtual keyword` from it and regenerated the templated code.

Once this builds I'm happy to merge it.

Reply:
@JimBobSquarePants @antonfirsov Ooof I was looking at the wrong overload 🤦‍♂️
I did have the feeling I was being stupid, sorry to have wasted your time on this!
Happy to hear the rest of the PR was ok though, thanks for merging it! 😊

Reply:
Haha no worries, easy  mistake to make!